Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -1445,10 +1445,19 @@ - (NSDictionary *) parseFieldDefinitionStringParts:(NSArray *)definitionParts
 		// Field defaults
 		} else if ([detailString isEqualToString:@"DEFAULT"] && (definitionPartsIndex + 1 < partsArrayLength)) {
 			detailParser = [[SPSQLParser alloc] initWithString:[definitionParts safeObjectAtIndex:definitionPartsIndex+1]];
-			if([[detailParser unquotedString] isEqualToString:@"NULL"])
-				[fieldDetails setObject:[NSNull null] forKey:@"default"];
-			else
-				[fieldDetails setValue:[detailParser unquotedString] forKey:@"default"];
+
+      // To know if default value is a function
+      [fieldDetails setValue:@NO forKey:@"isfunction"];
+      if([[detailParser unquotedString] isEqualToString:@"NULL"]) {
+        [fieldDetails setObject:[NSNull null] forKey:@"default"];
+      }else {
+        // Example values: function_name(), 'function_name()', 'invalid_function_name()', etc.
+        if ([detailParser isMatchedByRegex:SPFunctionNamePattern]) {
+          [fieldDetails setValue:@YES forKey:@"isfunction"];
+        }
+        
+        [fieldDetails setValue:[detailParser unquotedString] forKey:@"default"];
+      }
```

Comment:
This logic block is a bit unclear - suggest instead of having the nested if here to have multiple if/elseif/else conditions at the top level and avoid the nesting

Reply:
Actually, i've only added this if. The change is containing re-formatted code. I don't really expect to improve the whole logic for the parent block.

```c
        // Example values: function_name(), 'function_name()', 'invalid_function_name()', etc.
        if ([detailParser isMatchedByRegex:SPFunctionNamePattern]) {
          [fieldDetails setValue:@YES forKey:@"isfunction"];
        }
```

Reply:
You could also provide what is the better one, pseudo-code is ok. Thanks!